### DISCUSSION TITLE
[Stargazers] Graph type option

### DISCUSSION THREAD

**Discussion by Andre601** [lowlighter/metrics#812] [Ideas]
My suggestion is to add an option to the [`Stargazers`](https://github.com/lowlighter/metrics/blob/master/source/plugins/stargazers/README.md) plugin (And maybe by extension to other ones with similar graphs) that would allow to select the type of graph it should show.

As an example, could there be two options, alongside the default one which I simply call "bars" here:

- ### **Line**
  Displays the recent stars as a continuous line. ([Image Source](https://github.com/lowlighter/metrics/graphs/commit-activity))
  ![image](https://user-images.githubusercontent.com/11576465/151023521-a30a6402-d624-438a-9a84-5f5bc79d4a4e.png)

- ### **Filled-line**
  Similar to the Line option, but would have the bottom section filled. ([Image Source](https://github.com/lowlighter/metrics/graphs/contributors))
  ![image](https://user-images.githubusercontent.com/11576465/151023780-dd0a7daa-cee1-446c-bba6-7bd8fbe26665.png)

- ### **Add-remove**
  Not only shows recent starring per day, but also lost stars. May require to calculate a diff to see if there was any loss. ([Image Source](https://github.com/lowlighter/metrics/graphs/code-frequency))
  ![image](https://user-images.githubusercontent.com/11576465/151024124-8b36958c-7c99-4abc-984b-cad23516ee5b.png)

I can imagine the following plugins supporting this option:

- [Coding Habits](https://github.com/lowlighter/metrics/blob/master/source/plugins/habits/README.md)
- Maybe [Recent Activity](https://github.com/lowlighter/metrics/blob/master/source/plugins/activity/README.md)? Could have a graph showing increase/decrease of specific actions per day... Maybe a separate plugin for this?
- Maybe [GitHub Sponsors](https://github.com/lowlighter/metrics/blob/master/source/plugins/sponsors/README.md)? If it would support knowing current sponsors amount, have a graph show the increase/decrease? Maybe an option to show both new sponsors and earnings or just one of the two?

Those are only suggestions and I can understand them not being implemented soon, or even at all, since I can imagine the heavy work behind making those kinds of graphs work how they should.
So decide however you want. I accept any decision you make.

**Comment by lowlighter** [comment]
Hi 👋 !

Thanks a lot for your feedback! 
It's similar to #781 but yours extends actually a lot the idea and I really like what is proposed 👍 

Currently these plugins use a "div graphs" but with the introduction of [`💹 Stock prices`](https://github.com/lowlighter/metrics/blob/master/source/plugins/community/stock/README.md) plugins, [node-chartist](https://gionkunz.github.io/chartist-js/examples.html) was added to support more complex graphs, and all the ones you suggested seems to be supported so it would only be a matter of adding the options and connecting data with the chartist api. Shouldn't be too hard to implement.

The stargazers and coding habits can definitely support this.

For recent activity I don't think it'll be possible, the GitHub API doesn't actually returns that much events (about 200-400 events max I think) which can actually be only a few days for active users, so I don't think it is that revelant and worth the hassle

As for sponsors, I don't think it'll be implemented. From what I recall the GitHub API didn't provide that much data (I remember that even tiers list wasn't retrievable per user) and since it's also probably one of the least used plugin I don't feel like spending too much time on it.
Of course if someone else is motivated enough to include this I won't be against 🙂

**Reply by Andre601** [reply]
Nice to hear that parts of the rendering already exist.

Can also agree on the parts about recent activity and sponsors. Those were more a "nice to have" idea than anything else.

One other idea I have could be a graph to display the amount of opened and closed issues, opened, closed and merged PRs and opened and answered discussions. Not sure how those would go in terms of limits and/or tokens required. Never worked that much with the GitHub API outside of maybe the public events endpoint.
